cmd/geranos/s3uploader.go

```diff
@@ -47,29 +49,29 @@ const blobKeyPrefix = "containers/images/"
 const manifestKeyPrefix = "geranos/uploaded-images/"
 
 type s3Uploader struct {
-	svc            *s3.S3
-	uploader       *s3manager.Uploader
+	svc            *s3.Client
+	uploader       *manager.Uploader
 	reuploadLayers bool
 	dryRun         bool
 }
 
 func newS3Uploader(dryRun bool) (*s3Uploader, error) {
-	cfg := []*aws.Config{}
-	// force anonymous configs for dry run uploaders
-	if dryRun {
-		cfg = append(cfg, &aws.Config{
-			Credentials: credentials.AnonymousCredentials,
-		})
-	}
-	sess, err := session.NewSession(cfg...)
+	cfg, err := config.LoadDefaultConfig(context.TODO())
 	if err != nil {
 		return nil, err
 	}
+	if dryRun {
+		// Use anonymous credentials for dry run
+		cfg.Credentials = credentials.NewStaticCredentialsProvider("", "", "")
```

Comment:
This is probably the part i'd want to double check, we want to be sure this is anonymous and doesn't attempt to load credentials, it's not as clear in this iteration.

Reply:
fixed this after dry run testing